src/gps/GPS.cpp

```diff
@@ -563,24 +589,25 @@ int32_t GPS::runOnce()
         }
         GPSInitFinished = true;
         if (config.position.gps_enabled == false) {
-            doGPSpowersave(false);
-            return 0;
+            return disable();
         }
     }
     if (config.position.gps_enabled == false)
-        return 0;
+        return disable();
 
     // Repeaters have no need for GPS
-    if (config.device.role == meshtastic_Config_DeviceConfig_Role_REPEATER)
+    if (config.device.role == meshtastic_Config_DeviceConfig_Role_REPEATER) {
```

Comment:
Shouldn't this code be at the *beginning* of runOnce(). If GPS is disabled there shouldn't be any probing.